### PR TITLE
fix: Decimal support added in django spanner

### PR DESCRIPTION
**Pull request by vi3k6i5**
fix: Decimal support added in django spanner

### CODE REVIEW

File: tests/inspectdb/tests.py [hunk 1]
```diff
@@ -101,10 +101,8 @@ def test_number_field_types(self):
         assertFieldType('bool_field', "models.{}()".format(bool_field_type))
         assertFieldType('null_bool_field', 'models.{}(blank=True, null=True)'.format(bool_field_type))
 
-        if connection.features.can_introspect_decimal_field:
-            # django-spanner creates DecimalField as FLOAT64 until Spanner
-            # supports numeric.
-            assertFieldType('decimal_field', "models.FloatField()")
+        if connection.vendor != 'sqlite':
```
Comment (c24t) [inline tests/inspectdb/tests.py]: If we just rolled back the relevant hunk in d18d92245a4 this line wouldn't change, but it looks like this line comes from https://github.com/django/django/commit/e198beadad56d5fdd06756b097446b7046a84010 instead. I think this works, but this is now another change we need to cherry-pick if we update django.

File: tests/inspectdb/tests.py [hunk 2]
```diff
@@ -101,10 +101,8 @@ def test_number_field_types(self):
...
+            assertFieldType('decimal_field', "models.DecimalField(max_digits=6, decimal_places=1)")
```
Comment (c24t) [inline tests/inspectdb/tests.py:105]: Looks like I'm still seeing an error in `inspectdb.tests.InspectDBTestCase.test_number_field_types`:

```
======================================================================
FAIL: test_number_field_types (inspectdb.tests.InspectDBTestCase)
Test introspection of various Django field types
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../django/tests/inspectdb/tests.py", line 108, in test_number_field_types
    assertFieldType('decimal_field', "models.DecimalField(max_digits=6, decimal_places=1)")
  File ".../django/tests/inspectdb/tests.py", line 56, in assertFieldType
    self.assertEqual(definition, out_def)
AssertionError: 'mode[22 chars]gits=6, decimal_places=1)' != 'mode[22 chars]gits=10, decimal_places=5)  # max_digits and d[77 chars]loat'
- models.DecimalField(max_digits=6, decimal_places=1)
+ models.DecimalField(max_digits=10, decimal_places=5)  # max_digits and decimal_places have been guessed, as this database handles decimal fields as float
```

File: tests/model_forms/models.py [hunk 3]
```diff
@@ -212,7 +212,9 @@ def __str__(self):
 
 
 class Price(models.Model):
-    price = models.DecimalField(max_digits=10, decimal_places=2)
+    # Numeric field/Decimal Field is not supported in primary key/unique key.
+    # So changed it to Float field.
+    price = models.FloatField()
```
Comment (c24t) [inline tests/model_forms/models.py:217]: We may be asking for trouble continuing to patch tests on this fork instead of rewriting them in spanner-django.

**Review by c24t — COMMENTED** [review]
The changes look fine, but keeping this testing fork up to date with master while maintaining a bunch of misc test patches like this is going to be very painful. We should consider 

I tested this with https://github.com/googleapis/python-spanner-django/pull/620, but note that I had to un-disable validation for decimal fields:
https://github.com/vi3k6i5/python-spanner-django/blob/40a8fcecb0dc1da321567b91524d8d0e9e2273e1/django_spanner/validation.py#L28. Are you doing something different?

No problem merging this, I just want to make sure the tests pass first. The fewer commits on this fork that we have to keep up to date, the better.

File: tests/model_forms/models.py (hunk 3 above)
Comment (vi3k6i5) [reply tests/model_forms/models.py:217]: model migrations are applied by django test at the very start, so even if we skip a particular test this model migration causes a failure. Either we skip the entire test module or we do this overwrite, we don't have very good options here unit the feature is supported by Spanner:(

File: tests/inspectdb/tests.py (hunk 2 above)
Comment (vi3k6i5) [reply tests/inspectdb/tests.py:105]: I am seeing the same error, wanted to check with you, what should we do if someone tries to create a decimal field with custom precision and decimal places ? should we throw an error or overwrite to spanner precission and scale.

Comment (vi3k6i5) [reply tests/inspectdb/tests.py:105]: for now fixed it by changing the verification logic to use the same as sqllite.

File: tests/inspectdb/tests.py (hunk 1 above)
Comment (vi3k6i5) [reply tests/inspectdb/tests.py]: yes, this is added issue to maintain.

**Review by c24t — APPROVED** [review]

**Pull request merged**